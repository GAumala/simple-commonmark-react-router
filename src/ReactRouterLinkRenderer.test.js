const ReactRouterLinkRenderer = require('./ReactRouterLinkRenderer.js').default
const renderNodes = require('simple-commonmark-react').renderNodes

test('should render links with React Router\'s Link', () => {
  const options = { customRenderers: { link: ReactRouterLinkRenderer } }
  const markdown = 'Here is a [link](/some/path)'

  const nodes = renderNodes(markdown, options)
  expect(nodes.length).toEqual(1)

  const paragraph = nodes[0]
  expect(paragraph.props.children.length).toEqual(2)

  const link = paragraph.props.children[1]
  expect(link.props.to).toEqual('/some/path')
  expect(link.props.children).toEqual(['link'])
})
