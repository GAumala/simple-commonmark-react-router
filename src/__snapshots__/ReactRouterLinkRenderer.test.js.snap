// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render markdown correctly 1`] = `
<div>
  <p
    className={undefined}
  >
    Here is a 
    <a
      className={undefined}
      href="/some/path"
      onClick={[Function]}
    >
      link
    </a>
  </p>
</div>
`;
